File: lambdaconscarcdr.dart
#!/usr/bin/env dart

var kons  = (hd, tl) => (x) => x ? hd : tl;
var virst = (l) => l(true );
var rrest = (l) => l(false);

var lstt  = kons(1, kons(2, null));

virst(rrest(lstt));

File: lambdaconscarcdr.jl
#!/usr/bin/env julia

kons  = (hd, tl) -> x -> if x; hd else tl end
virst = l -> l(true )
rrest = l -> l(false)

lstt  = kons(1, kons(2, nothing))

virst(rrest(lstt))
